File: 1623.py
MAX = 10**9
n = 20
arr = [1 for i in range(20)]
best = MAX
for i in range(2**n):
    part2 = 0
    part1 = 0
    temp = i
    j = 0
    while(j < n):
        if temp&1<<j:
            part2 += arr[j]
        else:
            part1 += arr[j]
        j += 1
    best = min(best, abs(part2 - part1))

print(best)

File: 1623.cpp
#include<bits/stdc++.h>
using namespace std;
 
#define l long
#define ll long long
#define au auto
#define REP(i, a, n, s) for(au i=a; i<n; i+=s)
#define PB push_back
#define MP make_pair
#define umap unordered_map
 
ll pow(l n, l p){
	ll ans = 0, temp = n;
	while(p){
		if(p&1) ans*=temp;
		temp *= n;
		p >>= 1;
	}
	return ans;
 
}
 
int main(){
	l n, temp, j;
	cin >> n;
	l arr[n];
	REP(i, 0, n, 1) cin >> arr[i];
 
	l grp1, grp2, ans = 1e9;
	REP(i, 0, pow(2, n), 1){
		grp1 = 0; grp2 = 0;
		temp = i;
		REP(j, 0, n, 1){
			if(temp&1<<j) grp1 += arr[j];
			else grp2 += arr[j];
		}
		ans = min(ans, abs(grp1 - grp2));
	}
	cout << ans;
 
}
